client/src/controllers/InitController.test.js

```diff
@@ -148,4 +148,39 @@ describe('InitController', () => {
       ]);
     });
   });
+
+  describe('with a detail', () => {
+    const handleEvent = jest.fn();
+    document.addEventListener('w-init:ready', handleEvent);
+
+    beforeAll(() => {
+      jest.clearAllMocks();
+
+      application?.stop();
+
+      document.body.innerHTML = `
+        <div
+          id="test"
+          class="test-detail"
+          data-controller="w-init"
+          data-w-init-detail-value='{"someMessage": "some value"}'
+          Test body
+        </div>
+      `;
+
+      application = Application.start();
+    });
+
+    it('should dispatch event with a detail', async () => {
+      application.register('w-init', InitController);
+      await Promise.resolve(); // no delay, just wait for the next tick
+
+      expect(handleEvent).toHaveBeenCalledTimes(1);
+      const dispatchedEvent = handleEvent.mock.calls[0][0];
+      const detail = dispatchedEvent.detail;
+
+      // check if the event was dispatched with the correct detail object
+      expect(detail.someMessage).toBe('some value');
+    });
```

Comment:
Here's a refined test that checks both the 'init' event and a custom event, both getting the event detail as expected.

```suggestion
    it('should dispatch event with a detail', async () => {
      application.register('w-init', InitController);
      await Promise.resolve(); // no delay, just wait for the next tick

      expect(handleEvent).toHaveBeenCalledTimes(2);

      const [[firstEvent], [secondEvent]] = handleEvent.mock.calls;

      expect(firstEvent.type).toEqual('w-init:ready');
      expect(firstEvent.detail).toEqual(detail);

      expect(secondEvent.type).toEqual('my-custom:event');
      expect(secondEvent.detail).toEqual(detail);
    });
```